docs/integration_guide.md

```diff
@@ -26,31 +26,130 @@ The order of events for relaying transactions or deploying smart wallets through
+
+    const config = await resolveConfiguration(web3.currentProvider,
+        {
+            verbose: window.location.href.includes("verbose"),
+            onlyPreferredRelays: true,
+            preferredRelays: ["http://localhost:8090"],
+            factory: smartWalletFactoryAddress,
+            gasPriceFactorPercent: 0,
+            relayLookupWindowBlocks: 1e5,
+            chainId: 33,
+            relayVerifierAddress,
+            deployVerifierAddress,
+            smartWalletFactoryAddress
+        });
+        resolvedConfig.relayHubAddress = "0x3bA95e1cccd397b5124BcdCC5bf0952114E6A701"; // the relay hub contract address (can be retrieved from the summary of the deployment).
+
+    const provider = new RelayProvider(web3.currentProvider, config);
+    
+    provider.addAccount({
+        address: accountAddress,
+        privateKey: Buffer.from(accountPrivateKey.replaceAll("0x", ""), "hex") 
+    });
+
+    web3.setProvider(provider);
+    
+    const tokenContract = "0x0E569743F573323F430B6E14E5676EB0cCAd03D9"; // token address to use on smart wallet
+    const tokenAmount = "100"; // total token amount for the smart wallet, the smart wallet address should have more than this number before calling the deploy.
+
+    // deploy smart wallet
+    const deployTransaction = await provider.deploySmartWallet({
+        from: accountAddress,
+        to: ZERO_ADDRESS,
+        gas: "0x27100",
+        value: "0",
+        callVerifier: deployVerifierAddress,
+        callForwarder: smartWalletFactoryAddress,
+        tokenContract,
+        tokenAmount,
+        data: "0x",
+        index: smartWalletIndex,
+        recoverer: ZERO_ADDRESS,
+        isSmartWalletDeploy: true,
+        onlyPreferredRelays: true,
+        smartWalletAddress
+    });
+    
+    // relay transaction
+    const unsigned_tx = {
+        // some common web3 transaction with the common parameters.
+    };
+
+    const tokenAmountForRelay = "10";
+    
+    const relayTransaction = web3.eth.sendTransaction({
+        from: accountAddress,
+        callVerifier: relayVerifierAddress,
+        callForwarder: smartWalletAddress,
+        isSmartWalletDeploy: false,
+        onlyPreferredRelays: true,
+        tokenAmount: tokenAmountForRelay,
+        tokenContract,
+        ...unsigned_tx,
+    });
 ```
 
+Before running this example, you need to know a few requirements:
+
+1. The smart wallet address generated by the contract call should be funded with tokens before running the deploy call.
+2. The token address you use should be allowed, to do that you need to make a call to the contracts involved to allow
+   working with your token, only the owner of the contracts can do that, but if you are running this on regtest, then the account[0]
+   is the owner and you can allow tokens by calling the relay verifier and deploy verifier (for both wallets, smart wallet and custom smart wallet) contracts manually with web3.
```

Comment:
please break up this sentence into multiple ones, it's a bit hard to read as it is

Reply:
also, account[0] should be `accounts[0]`

Reply:
what if you are not running this on regtest? we should guide the user in this case too

Reply:
Should be the same for all chains, the only thing that changes is that in testnet and mainnet instead of you being allowing the tokens you have to use one of the already allowed tokens.